src/utils/scalloc.h

```diff
@@ -0,0 +1,14 @@
+#ifndef _SCHAUFEL_UTILS_SCALLOC_H
+#define _SCHAUFEL_UTILS_SCALLOC_H
+
+#include "utils/logger.h"
+
+void *scalloc(size_t n, size_t s, char *file, size_t line);
+#define SCALLOC(n, s) scalloc(( n), (s), __FILE__, __LINE__);
+
+#define SFREE(e) \
+    free(e); \
+    e = NULL;
```

Comment:
It's a good idea to wrap this into a `do { free... } while (0)` block to avoid potential side-effects.